examples/tree_2d_dgsem/elixir_kpp.jl

```diff
@@ -0,0 +1,152 @@
+
+# The KPP flux is F(u) = (sin(u), cos(u))
+@inline function Trixi.flux(u, orientation::Integer, ::KPPEquation2D)
+    if orientation == 1
+        return SVector(sin(u[1]))
+    else
+        return SVector(cos(u[1]))
+    end
+end
+
+# Since the KPP problem is a scalar equation, the entropy-conservative flux is uniquely determined
+@inline function Trixi.flux_ec(u_ll, u_rr, orientation::Integer, ::KPPEquation2D)
+    if abs(u_ll[1] - u_rr[1]) < 1e-12
+        return 0.5*(flux(u_ll, orientation, KPPEquation2D())+flux(u_rr, orientation, KPPEquation2D()))
+    else
+        factor = 1.0/(u_rr[1] - u_ll[1])
+        if orientation == 1
+            return SVector(factor*(-cos(u_rr[1]) + cos(u_ll[1])))
+        else
+            return SVector(factor*(sin(u_rr[1]) - sin(u_ll[1])))
+        end
+    end
+end
+
+# Wavespeeds
+const kpp_wavespeed = 1.0
+@inline Trixi.max_abs_speeds(u, ::KPPEquation2D) = (kpp_wavespeed, kpp_wavespeed)
+@inline Trixi.max_abs_speed_naive(u_ll, u_rr, orientation::Integer, ::KPPEquation2D) = kpp_wavespeed
+@inline Trixi.max_abs_speed_naive(u_ll, u_rr, normal_direction::AbstractVector, ::KPPEquation2D) = kpp_wavespeed * norm(normal_direction)
+
+# Compute entropy: we use the square entropy
+@inline Trixi.entropy(u::Real, ::KPPEquation2D) = 0.5 * u^2
+@inline Trixi.entropy(u, ::KPPEquation2D) = entropy(u[1], equation)
+
+# Convert between conservative, primitive, and entropy variables. The conserved quantity "u" is also
+# considered the "primitive variable". Since we use the square entropy, "u" is also the entropy
+# variable.
+@inline Trixi.cons2prim(u, ::KPPEquation2D) = u
+@inline Trixi.cons2entropy(u, ::KPPEquation2D) = u
+
+# Total energy
+@inline Trixi.energy_total(u::Real, ::KPPEquation2D) = 0.5 * u^2
+@inline Trixi.energy_total(u, equation::KPPEquation2D) = energy_total(u[1], equation)
+
+Trixi.varnames(::Any, ::KPPEquation2D) = ("u",)
+
+# Extract the solution as a scalar, for use in the indicator
+@inline cons_sol(u, ::KPPEquation2D) = u[1]
+
+# Standard KPP test problem with discontinuous initial condition
+function initial_condition_kpp(x, t, ::KPPEquation2D)
+    if x[1]^2 + x[2]^2 < 1
+        return SVector(0.25*14.0*pi)
+    else
+        return SVector(0.25*pi)
+    end
+end
+
+###############################################################################
+# semidiscretization of the KPP problem
+equation = KPPEquation2D()
+surface_flux = flux_lax_friedrichs
+volume_flux = flux_ec
+# Can also compare solution obtained without using entropy-conservative flux. This will not converge
+# to the correct (entropy) solution!
+# volume_flux = flux_central
+polydeg = 3
+basis = LobattoLegendreBasis(polydeg)
+shock_indicator = IndicatorHennemannGassner(equation, basis,
+                                            alpha_max=0.5,
+                                            alpha_min=0.001,
+                                            alpha_smooth=true,
+                                            variable=cons_sol)
```

Comment:
You can also write this as
```suggestion
                                            variable=first)
```
and avoid defining `cons_sol` above.

Reply:
Good idea, done.